### DISCUSSION TITLE
What is base for logprobs?

### DISCUSSION THREAD

**Discussion by wongjingping** [vllm-project/vllm#8766] [Q&A]
Hi there, I wanted to check what is the base for logprobs ie to get the normalized probability, do I do 2**(logprob) or 10**(logprob) or exp(logprob)?

**Comment by xiaobanni** [comment]
Also curious. Is there an answer now?

**Comment by wongjingping** [comment]
Based on this [line](https://github.com/vllm-project/vllm/blob/aa0dc77ef53b365ddf54be51748c166895a0bcd9/vllm/model_executor/layers/sampler.py#L295) it seems like it's using the natural log 
```py
logprobs = torch.log_softmax(logits, dim=-1, dtype=torch.float)
```
Based on the [pytorch docs](https://docs.pytorch.org/docs/stable/generated/torch.nn.LogSoftmax.html#torch.nn.LogSoftmax) the `log_softmax` performs the natural log.
